rust/chains/hyperlane-cosmos/src/providers/grpc.rs

```diff
@@ -271,6 +285,59 @@ impl WasmGrpcProvider {
         .map_err(Into::<HyperlaneCosmosError>::into)?;
         Ok(account)
     }
+
+    /// Injective-specific logic for querying an account.
+    async fn account_query_injective(&self, account: String) -> ChainResult<BaseAccount> {
+        let request = tonic::Request::new(
+            injective_std::types::cosmos::auth::v1beta1::QueryAccountRequest { address: account },
+        );
+
+        // Borrowed from the logic of `QueryAccountClient` in `cosmrs`, but using injective types.
+
+        let mut grpc_client = tonic::client::Grpc::new(self.channel.clone());
+        grpc_client
+            .ready()
+            .await
+            .map_err(Into::<HyperlaneCosmosError>::into)?;
+
+        let codec = tonic::codec::ProstCodec::default();
+        let path = http::uri::PathAndQuery::from_static("/cosmos.auth.v1beta1.Query/Account");
+        let mut req: tonic::Request<
+            injective_std::types::cosmos::auth::v1beta1::QueryAccountRequest,
+        > = request.into_request();
+        req.extensions_mut()
+            .insert(GrpcMethod::new("cosmos.auth.v1beta1.Query", "Account"));
+
+        let response: tonic::Response<
+            injective_std::types::cosmos::auth::v1beta1::QueryAccountResponse,
+        > = grpc_client
+            .unary(req, path, codec)
+            .await
+            .map_err(Into::<HyperlaneCosmosError>::into)?;
+
+        let mut eth_account = injective_protobuf::proto::account::EthAccount::parse_from_bytes(
```

Comment:
wondering what other consumers of `injective_protobuf` there are, would've expected there already is logic somewhere to convert `EthAccount` into `BaseAccount` :/

Reply:
sadly couldn't find any that would help out here